sympy/polys/matrices/domainmatrix.py

```diff
@@ -126,29 +345,272 @@ def __pow__(A, n):
+        r"""
+        Performs term by term multiplication for the second DomainMatrix
+        w.r.t first DomainMatrix. Returns a DomainMatrix whose rows are
+        list of DomainMatrix matrices created after term by term multiplication.
+
+        Parameters
+        ==========
+
+        A : Represents first DomainMatrix
+
+        B : Represents second DomainMatrix
+
+        Examples
+        ========
+
+        >>> from sympy import ZZ
+        >>> from sympy.polys.matrices import DomainMatrix
+        >>> domainMatrix1 = DomainMatrix([
+        ...    [ZZ(1), ZZ(2)],
+        ...    [ZZ(3), ZZ(4)]], (2, 2), ZZ)
+        >>> domainMatrix2 = DomainMatrix([
+        ...    [ZZ(1), ZZ(1)],
+        ...    [ZZ(0), ZZ(1)]], (2, 2), ZZ)
+
+        >>> domainMatrix1.mul(domainMatrix2)
+        DomainMatrix([[DomainMatrix([[1, 1], [0, 1]], (2, 2), ZZ),
+        DomainMatrix([[2, 2], [0, 2]], (2, 2), ZZ)],
+        [DomainMatrix([[3, 3], [0, 3]], (2, 2), ZZ),
+        DomainMatrix([[4, 4], [0, 4]], (2, 2), ZZ)]], (2, 2), ZZ)
+
+        See Also
+        ========
+
+        :py:meth:`~.matmul`
+
+        Returns
+        =======
+
+        DomainMatrix
+            term by term multiplication of two DomainMatrix matrices
+
+        """
         return A.from_rep(A.rep.mul(b))
 
     def matmul(A, B):
+        r"""
+        Performs matrix multiplication of two DomainMatrix matrices
+
+        Parameters
+        ==========
+
+        A : Represents first DomainMatrix
+
+        B : Represents second DomainMatrix
+
+        Examples
+        ========
+
+        >>> from sympy import ZZ
+        >>> from sympy.polys.matrices import DomainMatrix
+        >>> domainMatrix1 = DomainMatrix([
+        ...    [ZZ(1), ZZ(2)],
+        ...    [ZZ(3), ZZ(4)]], (2, 2), ZZ)
+        >>> domainMatrix2 = DomainMatrix([
+        ...    [ZZ(1), ZZ(1)],
+        ...    [ZZ(0), ZZ(1)]], (2, 2), ZZ)
+
+        >>> domainMatrix1.matmul(domainMatrix2)
+        DomainMatrix([[1, 3], [3, 7]], (2, 2), ZZ)
+
+        See Also
+        ========
+
+        :py:meth:`~.mul`
+
+        :py:meth:`~.pow`
+
+        Returns
+        =======
+
+        DomainMatrix
+            matrix multiplication of two DomainMatrix matrices
+
+        """
         return A.from_rep(A.rep.matmul(B.rep))
 
     def pow(A, n):
+        r"""
+        Perfoms multiplication on the given DomainMatrix for the given number of times.
+        Or else computes A^n
+
+        Parameters
+        ==========
+
+        A : Represents a DomainMatrix
+
+        n : Represents exponent for A
+
+        Examples
+        ========
+
+        >>> from sympy import ZZ
+        >>> from sympy.polys.matrices import DomainMatrix
+        >>> domainMatrix = DomainMatrix([
+        ...    [ZZ(1), ZZ(1)],
+        ...    [ZZ(0), ZZ(1)]], (2, 2), ZZ)
+
+        >>> domainMatrix.pow(2)
+        DomainMatrix([[1, 2], [0, 1]], (2, 2), ZZ)
+
+        See Also
+        ========
+
+        :py:meth:`~.matmul`
```

Comment:
The see also should be after returns, raises which should be with parameters:
https://numpydoc.readthedocs.io/en/latest/format.html